src/components/Common/OnboardingStatusBadge/index.tsx

```diff
@@ -0,0 +1,43 @@
+import type { ContractorOnboardingStatus1 } from '@gusto/embedded-api/models/components/contractor'
+import type { OnboardingStatus } from '@gusto/embedded-api/models/components/employee'
+import { useTranslation } from 'react-i18next'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+
+type OnboardingStatuses = ContractorOnboardingStatus1 | OnboardingStatus
+
+interface OnboardingStatusBadgeProps<T extends OnboardingStatuses> {
+  onboarded?: boolean
+  onboardingEntity: 'contractor' | 'employee'
+  onboardingStatus?: T | null
+}
+export const OnboardingStatusBadge = <T extends OnboardingStatuses>({
+  onboarded,
+  onboardingEntity,
+  onboardingStatus,
+}: OnboardingStatusBadgeProps<T>) => {
+  const { Badge } = useComponentContext()
+  const { t } = useTranslation()
+
+  //HACK: `never` should instead be a better type
```

Comment:
I feel the shame of this `never` type; I exceeded my timebox trying to type this nicely and figure we can get back to it later if we feel it's important